frontend/templates/product-list/ProductListDeviceNavigation.tsx

```diff
@@ -42,12 +42,12 @@ export function ProductListDeviceNavigation({
       >
          <SecondaryNavbarItem isCurrent>Parts</SecondaryNavbarItem>
          <SecondaryNavbarItem>
-            <NextLink href={guideUrl} passHref>
+            <NextLink href={encodeURI(guideUrl)} passHref>
```

Comment:
Same thing here. Instead of encoding the whole url, let's encode deviceHandle on line 22. In fact, we have a function encodeDeviceTitle that we should probably be using.